### ISSUE TITLE
Personal page broken for LDAP users after upgrade to 8.1

### ISSUE THREAD

**Issue by giancarlobi** [owncloud/core#17516]
### Steps to reproduce
1. Upgrade to 8.1
2. Login as LDAP user
3. Browse to Personal page
4. The page will be blank and this error appear into log:
```
Call to a member function getGID() on a non-object at \/var\/www\/owncloud\/settings\/personal.php#127
```
5.I have temporary reactivate the page editing the file /var/www/owncloud/settings/personal.php and
  a) comment out this line
```
$groups2 = array_map(function($group) { return $group->getGID(); }, $groups);
```
  b) added
```
foreach ($groups as $group) {
   if (is_object($group)) {
     $groups2[] = $group->getGID();
   }
}
```

6. It seems as the problem is an element of array $groups is not an object so line #127 return an error and break page rendering.

### Expected behaviour
Open personal page

### Actual behaviour
Page blank

### Server configuration
**Operating system**: 
Ubuntu server 14.04

**Web server:**
Apache/2.4.7 (Ubuntu)

**Database:**
MySQL

**PHP version:**
PHP 5.5.9-1ubuntu4.11

**ownCloud version:** (see ownCloud admin page)
ownCloud 8.1.0 (stable) 

**Updated from an older ownCloud or fresh install:**
Update from 8.0.4

**List of activated apps:**
 - activity: 2.0.1
 - files: 1.1.9
 - files_external: 0.2.3
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - gallery: 0.6.0
 - templateeditor: 0.1
 - user_ldap: 0.6.1
```

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => '***',
  'passwordsalt' => '****',
  'secret' => '****',
  'trusted_domains' =>
  array (
    0 => 'ocloud.to.cnr.it',
  ),
  'datadirectory' => '/srv/data',
  'overwrite.cli.url' => 'https://ocloud.to.cnr.it',
  'dbtype' => 'mysql',
  'version' => '8.1.0.8',
  'dbname' => '***',
  'dbhost' => '***',
  'dbtableprefix' => '***',
  'dbuser' => '***',
  'dbpassword' => '***',
  'installed' => true,
  'forcessl' => true,
  'ldapIgnoreNamingRules' => false,

  'theme' => '',
  'maintenance' => false,
  'singleuser' => false,
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

**Are you using an external user-backend, if yes which one:** 
LDAP+ActiveDirectory (W 2003)

#### LDAP configuration (delete this part if not used)

+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Configuration                 |                                                                                                                                                                                         |
+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                                                                                                                       |
| hasPagedResultSupport         |                                                                                                                                                                                         |
| homeFolderNamingRule          |                                                                                                                                                                                         |
| lastJpegPhotoLookup           | 0                                                                                                                                                                                       |
| ldapAgentName                 | ***                                                                                                                                              |
| ldapAgentPassword             | ***                                                                                                                                                                                     |
| ldapAttributesForGroupSearch  |                                                                                                                                                                                         |
| ldapAttributesForUserSearch   |                                                                                                                                                                                         |
| ldapBackupHost                | ***                                                                                                                                                                            |
| ldapBackupPort                | 389                                                                                                                                                                                     |
| ldapBase                      | DC=to,DC=cnr,DC=it                                                                                                                                                                      |
| ldapBaseGroups                | OU=UNR,DC=to,DC=cnr,DC=it                                                                                                                                                         |
| ldapBaseUsers                 | OU=UNR,DC=to,DC=cnr,DC=it                                                                                                                                                         |
| ldapCacheTTL                  | 600                                                                                                                                                                                     |
| ldapConfigurationActive       | 1                                                                                                                                                                                       |
| ldapEmailAttribute            | mail                                                                                                                                                                                    |
| ldapExperiencedAdmin          | 0                                                                                                                                                                                       |
| ldapExpertUUIDGroupAttr       | cn                                                                                                                                                                                      |
| ldapExpertUUIDUserAttr        | cn                                                                                                                                                                                      |
| ldapExpertUsernameAttr        |                                                                                                                                                                                         |
| ldapGroupDisplayName          | cn                                                                                                                                                                                      |
| ldapGroupFilter               | (&(|(objectclass=group))(|(cn=I_all)(cn=IPSP_all)))                                                                                                                                |
| ldapGroupFilterGroups         | I_all;IP_all                                                                                                                                                                     |
| ldapGroupFilterMode           | 0                                                                                                                                                                                       |
| ldapGroupFilterObjectclass    | group                                                                                                                                                                                   |
| ldapGroupMemberAssocAttr      | member                                                                                                                                                                                  |
| ldapHost                      | ***                                                                                                                                                                            |
| ldapIgnoreNamingRules         |                                                                                                                                                                                         |
| ldapLoginFilter               | (&(&(|(objectclass=user))(|(memberof=CN=osers,OU=IUfficioIT,OU=UNR,DC=to,DC=cnr,DC=it)))(|(uid=%uid)(|(mailPrimaryAddress=%uid)(mail=%uid))(|(sAMAccountName=%uid)))) |
| ldapLoginFilterAttributes     | sAMAccountName                                                                                                                                                                          |
| ldapLoginFilterEmail          | 1                                                                                                                                                                                       |
| ldapLoginFilterMode           | 0                                                                                                                                                                                       |
| ldapLoginFilterUsername       | 1                                                                                                                                                                                       |
| ldapNestedGroups              | 0                                                                                                                                                                                       |
| ldapNoCase                    | 1                                                                                                                                                                                       |
| ldapOverrideMainServer        | 0                                                                                                                                                                                       |
| ldapPagingSize                | 500                                                                                                                                                                                     |
| ldapPort                      | 389                                                                                                                                                                                     |
| ldapQuotaAttribute            | postOfficeBox                                                                                                                                                                           |
| ldapQuotaDefault              | 20 GB                                                                                                                                                                                   |
| ldapTLS                       | 0                                                                                                                                                                                       |
| ldapUserDisplayName           | displayname                                                                                                                                                                             |
| ldapUserFilter                | (&(|(objectclass=user))(|(|(memberof=CN=osers,OU=IUfficioIT,OU=UNR,DC=to,DC=cnr,DC=it)(primaryGroupID=15831))))                                                       |
| ldapUserFilterGroups          | ocloudUsers                                                                                                                                                                             |
| ldapUserFilterMode            | 0                                                                                                                                                                                       |
| ldapUserFilterObjectclass     | user                                                                                                                                                                                    |
| ldapUuidGroupAttribute        | auto                                                                                                                                                                                    |
| ldapUuidUserAttribute         | auto                                                                                                                                                                                    |
| turnOffCertCheck              | 0                                                                                                                                                                                       |
| useMemberOfToDetectMembership | 1                                                                                                                                                                                       |
+-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+



### Logs



#### ownCloud log (data/owncloud.log)
"Call to a member function getGID() on a non-object at \/var\/www\/owncloud\/settings\/personal.php#127"

**Comment by MorrisJobke** [comment]
> It seems as the problem is an element of array $groups is not an object so line #127 return an error and break page rendering.

cc @blizzz

**Comment by blizzz** [comment]
Found, fix will arrive shortly.

**Comment by blizzz** [comment]
Fix is here, please test: https://github.com/owncloud/core/issues/17516

**Comment by powerriegel** [comment]
Xould anyone tell me how I can apply this fix? I'm having the same issue but no LDAP backend but openmediavault plugin. My users come from openmediavault. I can still log in with any user but personal page is white. Log says

Call to a member function getGID() on a non-object at \/var\/www\/owncloud\/settings\/personal.php#127

**Comment by DeepDiver1975** [comment]
> openmediavault plugin

@powerriegel you should get in touch with the developer of that app - THX

**Comment by MorrisJobke** [comment]
It's merged for 8.1.1 #17704

**Comment by MorrisJobke** [comment]
@blizzz The one for master is still open

**Comment by blizzz** [comment]
Scusi!

**Comment by davitol** [comment]
:+1: Fixed "version":"8.1.0.8","versionstring":"8.1.0","edition":"
